src/libsyntax/parse/parser.rs

```diff
@@ -6341,7 +6341,11 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_item_const(&mut self, m: Option<Mutability>) -> PResult<'a, ItemInfo> {
-        let id = self.parse_ident()?;
+        let id = if self.eat_keyword(keywords::Underscore) {
+                    Ident::with_empty_ctxt(Symbol::gensym("_"))
+                } else {
+                    self.parse_ident()?
+                };
```

Comment:
Could you copy the logic from the current version of `fn parse_rename`, rather than from https://github.com/rust-lang/rust/pull/48922, it treat hygienic context more correctly.